base/dcp_common.go

```diff
@@ -82,11 +83,17 @@ type DCPCommon struct {
 	checkpointPrefix       string                         // DCP checkpoint key prefix
 }
 
-func NewDCPCommon(ctx context.Context, callback sgbucket.FeedEventCallbackFunc, metaStore Bucket, maxVbNo uint16, persistCheckpoints bool, dbStats *expvar.Map, feedID, checkpointPrefix string) *DCPCommon {
+func NewDCPCommon(ctx context.Context, callback sgbucket.FeedEventCallbackFunc, bucket Bucket, metaStore DataStore, maxVbNo uint16, persistCheckpoints bool, dbStats *expvar.Map, feedID, checkpointPrefix string) (*DCPCommon, error) {
```

Comment:
Should we switch bucket to CouchbaseBucketStore in the signature to avoid the AsCouchbaseStore check below?  Might be a better way to enforce type checking.

Reply:
I didn't get around to this after the test failures the other day, but could be done as follow-up.